androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/utils/TrackQualityItem.kt

```diff
@@ -0,0 +1,108 @@
+package com.profusion.androidenhancedvideoplayer.utils
+
+import android.os.Bundle
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.saveable.listSaver
+import androidx.media3.common.TrackGroup
+import androidx.media3.common.util.UnstableApi
+import com.google.common.collect.ImmutableList
+
+sealed interface TrackQualityItem
+
+@UnstableApi
+@Stable
+class TrackQualityAuto(
+    private val label: String
+) : TrackQualityItem {
+    override fun toString(): String {
+        return label
+    }
+    override fun equals(other: Any?): Boolean =
+        (other is TrackQualityAuto) &&
+            other.label == this.label
```

Comment:
The Android Studio is emitting the following warning:
```
Class has 'equals()' defined but does not define 'hashCode()' 
```
Maybe you need to implements `hashCode`.

Reply:
But I am wondering if is it better just use a `data class`? So you don't need to implements these comparison methods + toString.

Reply:
Oh sure, it is way better to use data class! Thanks!